### ISSUE TITLE
[BUG] Couriers don't take out dried kelp from restaurant's inventory

### ISSUE THREAD

**Issue by whisper2shade** [ldtteam/minecolonies#9782]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest MineColonies Version?

- [ ] I am running the latest **alpha** version of MineColonies for my Minecraft version.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### What were you playing at the time? Were you able to reproduce it in both settings?

- [x] Single Player
- [x] Multi Player

### Minecraft Version

1.20.1

### MineColonies Version

1.1.470-BETA

### Structurize Version

1.0.693-BETA

### Related Mods and their Versions

ATM9 modpack 0.2.43

### Current Behavior

Couriers won't take dried kelp from a restaurant to the mechanic to pack it for the stonesmelter to make charcoal.
At this point everything is stuck, because the restaurant inventory is full of dried kelp and a bunch of couriers are trying to fill it back with normal kelp, which won't fit.
I have disabled all (i think) the foods that could be kelp related (there is not a Dried Kelp food item), but still nothing. The proces is, that i use charcoal as fuel everywhere except stone smelter. Stone smelter uses kelp blocks to make charcoal and Restaurant dries the kelp.

### Expected Behavior

Dried kelp gets taken to the mechanic to make blocks and then to the smelter to make charcoal.

### Reproduction Steps

1. Have a restaurant lv. 3 with dried kelp recipe and charcoal as an allowed fuel source.
2. Have a mechanic with dried kelp block recipe
3. Have stone smelter with only dried kelp block as allowed fuel and have him make charcoal

### Logs

No relevant log entries found

### Anything else?

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by uecasm** [comment]
Dried kelp is itself a food.  Couriers don't in general like taking food away from the restaurant.  Though there ought to be an exception for things in a crafting chain; perhaps that's not checked at the right spot.

Another option might be to move dried kelp smelting to the stone smelter, although that might be weird for things like sushi where it is used as part of a food recipe.

In the meantime, I *think* you can fix the behaviour of your restaurant by disabling dried kelp in the food settings; couriers should be more willing to take it away in that case (for new requests).  Unless your modpack is changing things, there is indeed an entry for "dried kelp" in the Food settings.

**Comment by Thodor12** [comment]
We could always consider moving it to the planter as they are now also responsible for gathering kelp

**Comment by uecasm** [comment]
The planter isn't a furnace user.

Though another workaround is to use an external dried kelp farm (such as the one in the fortress style) that auto-smelts the kelp before it gets into the request system in the first place.

**Comment by Thodor12** [comment]
We can always give them that crafting module, doesn't the smelting AI have some kind of fallback in case of no furnaces in a schematic?

**Comment by uecasm** [comment]
Yeah, they don't do any work until they do have a furnace 😛 (and complain about it)

So no, that's not a viable option.

**Comment by MotionlessTrain** [comment]
There was another report as well, with baked potatoes requested from the postbox (with an assistant cook hired and the recipe taught). The cook stole the (baked) potatoes from the assistant cook, which resulted in all potatoes in the colony being baked, and none of them showing up in the postbox

Testing whether disabling baked potatoes did help was not possible anymore (and I haven't heard back from them later, when they did have potatoes again)

**Comment by uecasm** [comment]
Yeah, I think the Real Fix™ is to let the request system take away food that's part of a crafting chain; the other ideas are just workarounds.

Having said that, AFAIK the couriers currently should only ignore at most one stack of the item, so another possible workaround is to have more than one stack of dried kelp in the restaurant -- one will be kept there as food and any excess is allowed to be taken away for crafting.

**Comment by Raycoms** [comment]
The real fix is probably really untangling cook & assistant cook into separate buildings entirely. For starters, they could be small sub buildings with their own dedicated inventory.

**Comment by MotionlessTrain** [comment]
Yeah
Or (temporarily) making it an either/or could help (like the miner and quarrier can't work in the same building at the same time)

**Comment by uecasm** [comment]
The either/or thing wouldn't help, due to the way that the keep lists are structured.  Though that's not entirely insurmountable.

**Comment by Raycoms** [comment]
There are a couple of issues that go both way (assistant cook blocking normal cook and the other way around). These are only really solved by having their own dedicated inventories. And of course, the courier has to know about that.
